staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt.go

```diff
@@ -105,20 +121,158 @@ type EmptyHandler interface {
 	HandleEmpty()
 }
 
+// FairQueuingSystemImpl implements the FairQueuingSystem interface
+// TODO(aaron-prindle) remove this level of indirection and have
+// FQScheduler implement FairQueuingSystem
+type FairQueuingSystemImpl struct {
+	fqs *fq.FQScheduler
+}
+
+// NewFairQueuingSystem creates a new FairQueuingSystem
+func NewFairQueuingSystem(concurrencyLimit, desiredNumQueues, queueLengthLimit int, requestWaitLimit time.Duration, clk clock.Clock) FairQueuingSystem {
+	return &FairQueuingSystemImpl{
+		fqs: fq.NewFQScheduler(
+			concurrencyLimit,
+			desiredNumQueues,
+			queueLengthLimit,
+			requestWaitLimit,
+			clk,
+		),
+	}
+}
+
+// SetConfiguration sets the configuration for a FairQueuingSystem
+// SetConfiguration mutates the state of a FairQueuingSystem and handles
+// any update logic for changing fields (eg: adding queues)
+func (s *FairQueuingSystemImpl) SetConfiguration(concurrencyLimit, desiredNumQueues, queueLengthLimit int, requestWaitLimit time.Duration) {
+	s.fqs.SetConfiguration(concurrencyLimit, desiredNumQueues, queueLengthLimit, requestWaitLimit)
+}
+
+// Quiesce controls whether this system is quiescing.  Passing a
+// non-nil handler means the system should become quiescent, a nil
+// handler means the system should become non-quiescent.  A call
+// to Wait while the system is quiescent will be rebuffed by
+// returning `quiescent=true`.  If all the queues have no requests
+// waiting nor executing while the system is quiescent then the
+// handler will eventually be called with no locks held (even if
+// the system becomes non-quiescent between the triggering state
+// and the required call).r
+//
+// The filter uses this for a priority level that has become
+// undesired, setting a handler that will cause the priority level
+// to eventually be removed from the filter if the filter still
+// wants that.  If the filter later changes its mind and wants to
+// preserve the priority level then the filter can use this to
+// cancel the handler registration.
+func (s *FairQueuingSystemImpl) Quiesce(eh EmptyHandler) {
+	s.fqs.Lock.Lock()
+	defer s.fqs.Lock.Unlock()
+	if eh == nil {
+		s.fqs.Quiescent = false
+		return
+	}
+	// TODO(aaron-prindle) not really sure what to do for Quiesce...
+	// panic("not implemented")
```

Comment:
The other thing that needs to be done here is to check whether there are any requests queued or executing and if not then fork an invocation of the EmptyHandler.